paasta_tools/cli/cmds/logs.py

```diff
@@ -328,16 +359,37 @@ def get_log_reader():
 
 
 class LogReader(object):
-    def __init__(self, **kwargs):
-        pass
+    def __init__(self):
+        # Tailing, i.e actively viewing logs as they come in
+        self.SUPPORTS_TAILING = False
+        # Getting the last n lines of logs
+        self.SUPPORTS_LINE_COUNT = False
+        # Getting the last/prev n lines of logs from line #34013 for example
+        self.SUPPORTS_LINE_OFFSET = False
+        # Getting the logs between two given times
+        self.SUPPORTS_TIME = False
+
+        # Supporting at least one of these log retrieval modes is required
 
-    def tail_logs(service, levels, components, clusters, raw_mode=False):
+    def tail_logs(self, service, levels, components, clusters, raw_mode=False):
         raise NotImplementedError("tail_logs is not implemented")
 
+    def print_logs_by_time(self, service, start_time, end_time, levels, components, clusters, raw_mode):
+        raise NotImplementedError("print_logs_by_time is not implemented")
+
+    def print_last_n_logs(self, service, line_count, levels, components, clusters, raw_mode):
+        raise NotImplementedError("print_last_n_logs is not implemented")
+
+    def print_logs_by_offset(self, service, line_count, offset, levels, components, clusters, raw_mode):
+        raise NotImplementedError("print_logs_by_offset is not implemented")
+
 
 @register_log_reader('scribereader')
 class ScribeLogReader(LogReader):
-    def __init__(self, cluster_map, **kwargs):
+    def __init__(self, cluster_map):
```

Comment:
You are dropping the `**kwargs` here. Since the tests pass, I'm assuming this means that we weren't actually passing in any additional arguments (or that you've refactored that out).